### ISSUE TITLE
CRUD Modules Generator Module Naming Policy

### ISSUE THREAD

**Issue by yanhao-li** [area17/twill#71]
I improperly used a capitalized name as the module name in the command (php artisan twill:module Posts), and met the error of  
> Route [admin.posts.index] not defined.

It looks like the generator created Posts folders under views.admin

To avoid this kind of error,
I think maybe a detail naming spec documentation and further validation is needed for the CRUD module generate command:
`php artisan twill:module yourPluralModuleName` .

another approach is to convert the module name received directly to an appropriate name.

**Comment by ifox** [comment]
Definitely @yanhao-li. I agree just saying `yourPluralModuleName` as an example of using the CLI can be quite confusing, and the CLI output is not helping either.

Considering that Laravel relies heavily on pluralization, we should convert the input, I'm with you!

**Comment by IllyaMoskvin** [comment]
Sounds like in this case, capitalization – not pluralization – is what caused this error.

Casting to camel case as a first step might help when the module name is provided in pascal case.

That said, I've run into errors due to providing a singular module name before. It'd be great if it could accept both singular and plural module names as arguments. I'd love it if this:
```
php artisan twill:module yourPluralModuleName
```
...and this...
```
php artisan twill:module yourSingularModuleName
```
...both produced the same result.

For example, it should pluralize the module name when generating its table (e.g. `lightboxes`).

I believe it already singularizes the module's name when generating related tables:  
https://github.com/area17/twill/blob/bd84f41/src/Commands/ModuleMake.php#L98-L102
https://github.com/area17/twill/blob/16ec480/src/Commands/stubs/migration.stub#L30-L46

**Comment by yanhao-li** [comment]
@IllyaMoskvin exactly, I'm preparing a fix on my fork, should create a pull request shortly once I'm done.

**Comment by Paluschki** [comment]
Just wanted to mention that I also had problems with underscore and pluralization with other languages than english.

In german _sonderangebote_ is plural and I got an error when I migrated, because somewhere the logic appended an 's' to the name. Then I switchted to the english translation _special_offers_. Migration did work, but later I got a problems with a dependency (somehow the underscore broke when importing special_offersRepository).

It may be the case that there is well-known naming convention for tables, but I did not know of it until now. I think some clarification in the documentation would help future Twill users very much.
